src/Uno.UI/UI/Xaml/Controls/WebView/UnoWKWebView.iOSmacOS.cs

```diff
@@ -838,6 +838,21 @@ public override void DidFailProvisionalNavigation(WKWebView webView, WKNavigatio
 					}
 				}
 			}
+
+			public override void DecidePolicy(WKWebView webView, WKNavigationAction navigationAction, WKWebpagePreferences preferences, Action<WKNavigationActionPolicy, WKWebpagePreferences> decisionHandler)
```

Comment:
Should we be invoking back this method:

https://github.com/unoplatform/uno/blob/574a441497e0e6faee8d389a3dbac1c40d4d7891/src/Uno.UI/UI/Xaml/Controls/WebView/UnoWKWebView.iOSmacOS.cs#L654

It's curious that we have not seen this before on iOS.

To limit the impact, I'd suggest to keep this method macOS specific for now.

It may override/interact the behavior for the other ?

Reply:
I'm just starting out with the Apple world. Based on the documentation found [here](https://developer.apple.com/documentation/webkit/wknavigationdelegate/3223382-webview?language=objc), if it is not implemented, the runtime calls [webView: decidePolicyForNavigationAction: decisionHandler :](https://developer.apple.com/documentation/webkit/wknavigationdelegate/1455641-webview?language=objc).
I tried to copy the [DecidePolicy](https://github.com/unoplatform/uno/blob/74dc3e1be201a377cfa1a84d97928dc9e4dd55ff/src/Uno.UI/UI/Xaml/Controls/WebView/UnoWKWebView.iOSmacOS.cs#L654) logic to this overload and the problem persists. So the problem is in DecidePolicy which does not implement the specific case, not for the missing of the overaide of the olverload.

Reply:
I'm just starting out with the Apple world. Based on the documentation found [here](https://developer.apple.com/documentation/webkit/wknavigationdelegate/3223382-webview?language=objc), if it is not implemented, the runtime calls [webView: decidePolicyForNavigationAction: decisionHandler :](https://developer.apple.com/documentation/webkit/wknavigationdelegate/1455641-webview?language=objc).
I tried to copy the [DecidePolicy](https://github.com/unoplatform/uno/blob/74dc3e1be201a377cfa1a84d97928dc9e4dd55ff/src/Uno.UI/UI/Xaml/Controls/WebView/UnoWKWebView.iOSmacOS.cs#L654) logic to this overload and the problem persists. So the problem is in DecidePolicy which does not implement the specific case, not for the missing of the overaide of the olverload.